backend/endpoints/rooms.py

```diff
@@ -1,3 +1,121 @@
-from fastapi import APIRouter
+import requests
+from fastapi import APIRouter, BackgroundTasks, status
+from fastapi.responses import JSONResponse
+from schema.room import Room
+from utils.centrifugo_handler import Events, centrifugo_client
+from utils.db_handler import DB
+from utils.utility import get_rooms, sidebar_emitter
 
 router = APIRouter()
+
+
+async def extra_room_info(room_data: dict):
+    """provides some extra room information to be displayed on the sidebar
+    Args:
+        room_data {dict}: {object of newly created room}
+    Returns:
+        {dict}
+
+    """
+
+    if room_data["plugin_name"] == "channels":
+        return {
+            "category": "channel",
+            "group_name": "channel",
+            "room_name": room_data["room_name"],
+        }
+    return {"category": "direct messagine", "group_name": "dm", "room_name": None}
+
+
+async def centrifugo_publish(org_id, room_data, member_id):
+    """update new room to user's sidebar
+    Args:
+        org_id (str): id of organisation
+        room_data: data to be published to centrifugo
+        member_id: id of room_creator
+    Returns:
+        HTTP_200_OK: sidebar successfully updated
+        HTTP_424_FAILED_DEPENDENCY: update unsuccessful
+    """
+
+    other_info = await extra_room_info(room_data)
+    response_output = await sidebar_emitter(
+        org_id=DB.organization_id,
+        member_id=member_id,
+        category=other_info["category"],
+        group_name=other_info["group_name"],
+    )
+    try:
+        centrifugo_data = await centrifugo_client.publish(
+            room=f"{org_id}_{member_id}_sidebar",
+            event=Events.SIDEBAR_UPDATE,
+            data=response_output,
```

Comment:
The response output of the centrifugo function would already fields such as data, event, plugin id which would create duplicate data. Let data field only contain specific things you are sending aside from event, and plugin_id whcih should not be included.

Reply:
When I was researching on daemon threads and background task, I found you had to run them as a function inside the main function.
so, that is what this function is doing. Just publishing the new room to the sidebar. Maybe it's the naming. I will change it to a more descriptive name. 
or do you have some other ideas?